HeinOnline.js

```diff
@@ -0,0 +1,158 @@
+{
+	"translatorID": "3dcbb947-f7e3-4bbd-a4e5-717f3701d624",
+	"label": "HeinOnline",
+	"creator": "Frank Bennett",
+	"target": "^https?://heinonline\.org/HOL/(?:LuceneSearch|Page)\\?",
+	"minVersion": "1.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2015-07-04 10:32:10"
+}
+
+function detectWeb (doc, url) {
+	if (url.indexOf("/LuceneSearch?") > -1) {
+		return "multiple";
+	} else {
+		return "journalArticle";
+	}
+}
+
+function Data(doc) {
+	this.node = ZU.xpath(doc, '//form[@id="Print1"]');
+	var urlLst = doc.location.href.split('/');
+	this.urlbase = urlLst.slice(0, 4).join('/') + "/PDFsearchable?";
+	this.queryElems = [];
+	this.tail = "&sectioncount=1&ext=.pdf&nocover=";
+}
+
+Data.prototype.getval = function(name, returnOnly) {
+	var val = '';
+	var input = ZU.xpath(this.node, './/input[@name="' + name + '"]');
+	if (input && input.length) {
+		val = input[0].value ? encodeURIComponent(input[0].value) : '';
+	}
+	if (!returnOnly) {
+		this.queryElems.push(name + "=" + val);
+	}
+	return val;
+}
+
+Data.prototype.dump = function() {
+	return this.urlbase + this.queryElems.join("&") + this.tail;
+}
+
+function getSearchResults(doc) {
+	var results = ZU.xpath(doc, '//div[contains(@class, "lucene_search_result_b")]'),
+		items = {},
+		found = false
+	for (var i=0; i<results.length; i++) {
+		
+		var url = ZU.xpath(results[i], './/a[contains(@href, "Print")]');
+		url = (url && url.length) ? url[0].href : false;
+
+		var title = ZU.trimInternal(ZU.xpath(results[i], './/a[1]')[0].textContent);		
+		title = title.replace(/\s*\[[^\]]*\]$/, '');
+
+		if (!title || !url) continue;
+		url = url.replace(/Print/, "Page");
+		url = url.replace(/&terms=[^&]*/, '');
+		
+		items[url] = title;
+		found = true;
+	}
+	return found ? items : false;
+}
+
+function scrapePage(doc) {
+	var pdfPageURL = doc.location.href.replace(/\/Page\?/, "/Print?");
+	var item = new Zotero.Item();
+	var spans = ZU.xpath(doc, '//span[contains(@class, " Z3988") or contains(@class, "Z3988 ") or @class="Z3988"][@title]');
```

Comment:
The expression was lifted from the COinS translator. I assumed that xpath was used there because of the tag-name and title-attribute constraints.

Could get all elements with the Z3988 class, then discard any that are not spans or that have no title value. Not sure which is less attractive - leaving it like this for now.